internal/pkg/configmanager/manager.go

```diff
@@ -0,0 +1,271 @@
+package configmanager
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"go.uber.org/multierr"
+	"gopkg.in/yaml.v3"
+
+	"github.com/devstream-io/devstream/pkg/util/log"
+)
+
+type Manager struct {
+	ConfigFile string
+}
+
+func NewManager(configFileName string) *Manager {
+	return &Manager{
+		ConfigFile: configFileName,
+	}
+}
+
+// ConfigRaw is used to describe original raw configs read from files
+type ConfigRaw struct {
+	VarFile           string             `yaml:"varFile"`
+	ToolFile          string             `yaml:"toolFile"`
+	AppFile           string             `yaml:"appFile"`
+	TemplateFile      string             `yaml:"templateFile"`
+	PluginDir         string             `yaml:"pluginDir"`
+	State             *State             `yaml:"state"`
+	Tools             []Tool             `yaml:"tools"`
+	AppsInConfig      []AppInConfig      `yaml:"apps"`
+	PipelineTemplates []PipelineTemplate `yaml:"pipelineTemplates"`
+	Vars              map[string]any     `yaml:"-"`
+}
+
+// LoadConfig reads an input file as a general config.
+// It will return "non-nil, err" or "nil, err".
+func (m *Manager) LoadConfig() (*Config, error) {
```

Comment:
This function is too big now with c.a. 90 lines of code. I need to scroll up and down in my IDE to have a global understanding of it. Maybe we can extract some functions, for example, one major step as one function, so that this function could be much shorter, like:

```go
func LoadConfig() {
  // ...
  cis, cds = renderxxx()
  // ...
  constructApps()
  // ...
}
```